cql3/statements/index_target.cc

```diff
@@ -48,8 +48,10 @@ index_target::target_type index_target::from_sstring(const sstring& s)
         return index_target::target_type::keys;
     } else if (s == "entries") {
         return index_target::target_type::keys_and_values;
+    } else if (s == "regular_values") {
+        return index_target::target_type::regular_values;
     } else if (s == "values") {
-        return index_target::target_type::values;
+        return index_target::target_type::collection_values;
```

Comment:
Could we have a backward compatibility problem here?
If the "index target" is stored as a string in the schema (is it? I forgot?), then the value "values" used to mean one thing (index of a regular column) and after this patch means something else (collection values).
In other words the question is can we create a table with an index on a regular column before this patch, then upgrade to Scylla with this patch, and use this index correctly.

Reply:
It is stored as string in the schema table, however it wasn't possible before to actually store `values(some_column)`. I standardized the approach - `values` means collection values, omitting it entirely means that this is a regular column.

Reply:
Before this patch it was just `column_name`. Now, to clear the confusion:
```
cqlsh:ks> create table tt(pk int, reg int, m map<int, int>, primary key (pk));
cqlsh:ks> create index on tt(reg);
cqlsh:ks> create index on tt(keys(m));
cqlsh:ks> create index on tt(values(m));
cqlsh:ks> create index on tt(entries(m));
cqlsh:ks> select * from system_schema.indexes where keyspace_name = 'ks' and table_name = 'tt';

 keyspace_name | table_name | index_name       | kind       | options
---------------+------------+------------------+------------+--------------------------
            ks |         tt | tt_m_entries_idx | COMPOSITES | {'target': 'entries(m)'}
            ks |         tt |    tt_m_keys_idx | COMPOSITES |    {'target': 'keys(m)'}
            ks |         tt |  tt_m_values_idx | COMPOSITES |  {'target': 'values(m)'}
            ks |         tt |       tt_reg_idx | COMPOSITES |        {'target': 'reg'}
```

Reply:
My (second) question was what happens if you have (I didn't test):
```
create table tt(pk int, "keys(reg)" int);
create index on tt("keys(reg)");
```
Arguably this is an unreasonable example, but it should work - I create a column with the silly name "keys(reg)" and should be able to index that silly name. It's not a keys of anything - it's just the name of a regular column.

Reply:
And indeed it fails miserably. But it also was the case before collection indexes were introduced. Tested on `4.6.3`:
```
cqlsh> create table tab(pk int, "entries(m)" int, "cosdziwnego(m)" int, "values(l)" list<int>, regg int, listt list<int>, primary key(pk));
InvalidRequest: Error from server: code=2200 [Invalid query] message="No keyspace has been specified. USE a keyspace, or explicitly specify keyspace.tablename"
cqlsh> use ks;
cqlsh:ks> create table tab(pk int, "entries(m)" int, "cosdziwnego(m)" int, "values(l)" list<int>, regg int, listt list<int>, primary key(pk));
cqlsh:ks> create index on tab("entries(m)");
ConfigurationException: Unable to parse targets for index tab_entries(m)_idx (entries(m)): std::runtime_error (Column m not found)
cqlsh:ks> create index on tab("cosdziwnego(m)");
cqlsh:ks> create index on tab("values(l)");
InvalidRequest: Error from server: code=2200 [Invalid query] message="Cannot create secondary index on non-frozen collection or UDT column values(l)"
cqlsh:ks> create index on tab(regg);
cqlsh:ks> create index on tab(listt);
InvalidRequest: Error from server: code=2200 [Invalid query] message="Cannot create secondary index on non-frozen collection or UDT column listt"
cqlsh:ks> select * from system_schema.indexes;

 keyspace_name | table_name | index_name             | kind       | options
---------------+------------+------------------------+------------+------------------------------
            ks |        tab | tab_cosdziwnego(m)_idx | COMPOSITES | {'target': 'cosdziwnego(m)'}
            ks |        tab |           tab_regg_idx | COMPOSITES |           {'target': 'regg'}

(2 rows)
```

Perhpas a fix to it should be in a separate patch?

Meanwhile Cassandra:
```
cqlsh:ks> create table tab(pk int, "entries(m)" int, "cosdziwnego(m)" int, "values(l)" list<int>, regg int, listt list<int>, primary key(pk));
cqlsh:ks> create index on tab("entries(m)");
cqlsh:ks> create index on tab("cosdziwnego(m)");
cqlsh:ks> create index on tab("values(l)");
cqlsh:ks> create index on tab(regg);
cqlsh:ks> create index on tab(listt);
cqlsh:ks> select * from system_schema.indexes;

 keyspace_name | table_name | index_name           | kind       | options
---------------+------------+----------------------+------------+-----------------------------------
            ks |        tab | tab_cosdziwnegom_idx | COMPOSITES |    {'target': '"cosdziwnego(m)"'}
            ks |        tab |     tab_entriesm_idx | COMPOSITES |        {'target': '"entries(m)"'}
            ks |        tab |        tab_listt_idx | COMPOSITES |       {'target': 'values(listt)'}
            ks |        tab |         tab_regg_idx | COMPOSITES |                {'target': 'regg'}
            ks |        tab |      tab_valuesl_idx | COMPOSITES | {'target': 'values("values(l)")'}

(5 rows)
```

Reply:
If you prefer to improve this casase in a followup, let me know and I'll create a separate issue for it.

Reply:
Opened https://github.com/scylladb/scylla/issues/10707, so we can commit this PR and resolve this edge case later.

Reply:
If there is a backward compatibility problem, we should fix it now and not later. Otherwise we block creating a new branch.